sn_networking/src/lib.rs

```diff
@@ -146,6 +152,40 @@ pub fn sort_peers_by_key<'a, T>(
 
     Ok(sorted_peers)
 }
+/// Only return peers closer to key than the provided distance
```

Comment:
but the `provided distance` was calculated according to `the node's close range` ?
which could be way smaller than key's actuall close range, when `key` is in a far bucket?

Reply:
Hmm, what would you suggest here?

Reply:
well, in general, I think trying to use `our close range` for a `far bucket key`, especially based on `local RT peers` will deliver a wrong result with high chance.

for any key that belongs to a `full bucket`, all peers of that bucket shall be contacted for `further closer nodes` . 
but this will then very likely to `re-write the libp2p's interative closeness queries` .